server/src/main/kotlin/com/kamelia/hedera/core/response/DTO.kt

```diff
@@ -0,0 +1,69 @@
+@file:UseSerializers(UUIDSerializer::class)
+
+package com.kamelia.hedera.core.response
+
+import com.kamelia.hedera.rest.core.DTO
+import com.kamelia.hedera.util.UUIDSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+
+
+@Serializable
+data class MessageKeyDTO(
+    val key: String,
+    val parameters: Map<String, MessageKeyDTO>? = null,
+) : DTO {
+    constructor(key: String, vararg parameters: Pair<String, MessageKeyDTO>) : this(key, mapOf(*parameters))
+
+    fun withParameters(vararg parameters: Pair<String, String>): MessageKeyDTO {
+        val params = parameters.map { (k, v) -> k to MessageKeyDTO(v) }.toTypedArray()
+        this.parameters?.let { return copy(parameters = it.plus(params)) }
+        return copy(parameters = mapOf(*params))
+    }
+}
+
+fun String.asMessage() = MessageKeyDTO(this)
+fun String.asMessage(parameters: Map<String, String>? = null) =
+    MessageKeyDTO(this, parameters?.mapValues { MessageKeyDTO(it.value) })
+
+fun String.asMessage(vararg parameters: Pair<String, String>) =
+    MessageKeyDTO(this, mapOf(*(parameters.map { (k, v) -> k to MessageKeyDTO(v) }.toTypedArray())))
```

Comment:
use List<Pair<?, ?>>.toMap(), or even the `associate` function instead of this convoluted thing. Same in the withParameters method above